src/persist-client/src/write.rs

```diff
@@ -454,11 +454,16 @@ where
         let desc = Description::new(lower, upper, since);
 
         let (mut parts, mut num_updates) = (Vec::new(), 0);
+        let mut runs = vec![];
         for batch in batches.iter() {
             let () = validate_truncate_batch(&batch.batch.desc, &desc)?;
-            parts.extend_from_slice(&batch.batch.parts);
+            for run in batch.batch.runs() {
+                parts.extend_from_slice(run);
+                runs.push(parts.len());
+            }
             num_updates += batch.batch.len;
         }
+        runs.pop(); // We mark the end of each run above, but the last one is implicit
```

Comment:
[elsewhere in the code](https://github.com/MaterializeInc/materialize/blob/9aa1febad6136091f789bc7c8f4d10f79eaedaef/src/persist-client/src/internal/state.rs#L300-L306) the implicit run is considered the first one, so it's confusing me a bit here to reference it as the last one. either mental model works, but it's a bit of added trickiness